### PR TITLE
feat: Support configuring /PipeSpec/ErrorHandler

### PR DESCRIPTION
**Pull request by igarashitm**
Fixes: #188

### CODE REVIEW

**Review by lordrip — APPROVED** [review]

File: packages/ui/src/stubs/PipeErrorHandler.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
+export const pipeErrorHandlerJson = {
+  $schema: 'http://json-schema.org/draft-07/schema#',
```
[the remark this answers is not in the document]
Comment (igarashitm) [reply packages/ui/src/stubs/PipeErrorHandler.ts:2]: Yes the actual schema, we can remove this if we can use schema provider in tests

File: packages/ui/src/components/MetadataEditor/PipeErrorHandlerEditor.test.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+          model = m;
+        }}
+        schema={pipeErrorHandlerJson}
+      />,
+    );
+    const button = container.querySelector('button');
+    fireEvent(button!, new MouseEvent('click', { bubbles: true }));
+    const buttons = container.querySelectorAll('button');
+    buttons.forEach((button) => {
+      if (button.querySelector('span')?.querySelector('span')?.innerHTML === 'Log ErrorHandler') {
```
[the remark this answers is not in the document]
Comment (igarashitm) [reply packages/ui/src/components/MetadataEditor/PipeErrorHandlerEditor.test.tsx:45]: Any trick to do that when PatternFly generates the HTML? The `<Select>` generates `<button><span><span>` and I don't see a way to put the testid

File: packages/ui/src/components/MetadataEditor/PipeErrorHandlerEditor.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  schema: unknown;
+  model: unknown;
+  onChangeModel: (model: PipeErrorHandlerType) => void;
+}
+
+export const PipeErrorHandlerEditor: FunctionComponent<PropsWithChildren<PipeErrorHandlerEditorProps>> = (props) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [errorHandlerType, setErrorHandlerType] = useState<string>('none');
+  const schema = useMemo(() => props.schema as Record<string, unknown>, [props.schema]);
+  const schemaOneOf = useMemo(() => schema.oneOf as Record<string, unknown>[], [schema]);
```
[the remark this answers is not in the document]
Comment (igarashitm) [reply packages/ui/src/components/MetadataEditor/PipeErrorHandlerEditor.tsx:16]: not sure, just following the rule "values which could be cached across rerendering", any side effects here?

File: packages/ui/src/stubs/PipeErrorHandler.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (lordrip) [reply packages/ui/src/stubs/PipeErrorHandler.ts:2]: Ah, okeydokey, then let's keep it, as we need to work a bit around the schemas soon.

File: packages/ui/src/components/MetadataEditor/PipeErrorHandlerEditor.test.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (lordrip) [reply packages/ui/src/components/MetadataEditor/PipeErrorHandlerEditor.test.tsx:45]: Usually, they provide roles, like a `select` role: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/select_role

File: packages/ui/src/components/MetadataEditor/PipeErrorHandlerEditor.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (lordrip) [reply packages/ui/src/components/MetadataEditor/PipeErrorHandlerEditor.tsx:16]: in this case, there's no benefit, as we're using what comes from the parent as props, meaning that if this value change, the component will be rendered again, so no real gain here.

**Pull request merged**